src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> last = tail;
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        isIndexValid(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNode(index);
+        Node<T> previous = node.previous;
+        Node<T> newNode = new Node<>(previous, value, node);
+        node.previous = newNode;
+        if (previous == null) {
+            head = newNode;
+        } else {
+            previous.next = newNode;
+        }
+        size++;
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
+    }
+
+    public Node<T> checkIndex(int index) {
+        if (0 <= index && index < size) {
+            return getNode(index);
```

Comment:
you shouldn't call `getNode()` here, `checkIndex` only purpose is to check the index and throw an exception if the index isn't valid, also you don't `if else`, `if` will be enough